Zed crashes when doing "ciq" in vim mode

### Summary

<!-- Begin your issue with a one sentence summary -->
When I executed the Vim command 'ciq' (change inside quotes) Zed panicked

### Description
<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1. Turn on Vim mode
2. Create a *.sh file with the text  `'{"123"}'`
3. Enter normal mode
4. Navigate the cursor onto the number `2`
5. Type `ciq`

Actual Behavior:
Zed exits unexpectedly.

Expected Behavior:
Zed deletes `123` and then enters insert mode, placing the cursor at the location where the deleted text was.


### Zed Version and System Specs

Zed: v0.182.11 (Zed) 
OS: Linux X11 ubuntu 24.04
Memory: 62.5 GiB
Architecture: x86_64
GPU: NVIDIA GeForce RTX 4090 || NVIDIA || 570.86.10

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Paste your log inside the code block. -->
```log
2025-04-22T18:30:18.124651423+03:00 [INFO] ========== starting zed ==========
2025-04-22T18:30:18.139437695+03:00 [INFO] Compositor detection: _NET_WM_CM_S?=false, _NET_WM_CM_OWNER=false, _NET_SUPPORTING_WM_CHECK=true
2025-04-22T18:30:18.1394781+03:00 [INFO] x11: compositor present: true, gtk_frame_extents_supported: false
2025-04-22T18:30:18.172574315+03:00 [INFO] Presentation support: "VK_KHR_wayland_surface"
2025-04-22T18:30:18.172605814+03:00 [INFO] Presentation support: "VK_KHR_xcb_surface"
2025-04-22T18:30:18.172622174+03:00 [INFO] Presentation support: "VK_KHR_xlib_surface"
2025-04-22T18:30:18.172639536+03:00 [INFO] Enabling Vulkan Portability
2025-04-22T18:30:18.172654434+03:00 [INFO] Enabling color space support
2025-04-22T18:30:18.19305273+03:00 [INFO] Adapter: "NVIDIA GeForce RTX 4090"
2025-04-22T18:30:18.193087234+03:00 [INFO] Ray tracing is supported
2025-04-22T18:30:18.328870501+03:00 [INFO] Try connect ibus
2025-04-22T18:30:18.329419551+03:00 [INFO] perform;
2025-04-22T18:30:18.329448034+03:00 [INFO] authenticate;
2025-04-22T18:30:18.32948331+03:00 [INFO] read_command;
2025-04-22T18:30:18.329605397+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:18.329635463+03:00 [INFO] Opening main db
2025-04-22T18:30:18.330032842+03:00 [INFO] new;
2025-04-22T18:30:18.330093504+03:00 [INFO] socket reader;
2025-04-22T18:30:18.330121266+03:00 [INFO] read_socket;
2025-04-22T18:30:18.330181127+03:00 [INFO] read_socket;
2025-04-22T18:30:18.33031712+03:00 [INFO] read_socket;
2025-04-22T18:30:18.330422696+03:00 [INFO] read_socket;
2025-04-22T18:30:18.330497265+03:00 [INFO] read_socket;
2025-04-22T18:30:18.330525638+03:00 [INFO] Opening main db
2025-04-22T18:30:18.330667712+03:00 [INFO] read_socket;
2025-04-22T18:30:18.330757349+03:00 [INFO] read_socket;
2025-04-22T18:30:18.330823241+03:00 [INFO] read_socket;
2025-04-22T18:30:18.33232299+03:00 [INFO] Using git binary path: None
2025-04-22T18:30:18.351143502+03:00 [INFO] No prompt template overrides directory found at /home/pav/.local/share/zed/prompt_overrides. Using built-in prompts.
2025-04-22T18:30:18.351921328+03:00 [INFO] extensions updated. loading 12, reloading 0, unloading 0
2025-04-22T18:30:18.353249968+03:00 [INFO] activate is not implemented on Linux, ignoring the call
2025-04-22T18:30:18.355239778+03:00 [INFO] Opening main db
2025-04-22T18:30:18.35542925+03:00 [INFO] perform;
2025-04-22T18:30:18.355448896+03:00 [INFO] authenticate;
2025-04-22T18:30:18.355470537+03:00 [INFO] read_command;
2025-04-22T18:30:18.355534165+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:18.355863207+03:00 [INFO] socket reader;
2025-04-22T18:30:18.355890788+03:00 [INFO] read_socket;
2025-04-22T18:30:18.355936142+03:00 [INFO] read_socket;
2025-04-22T18:30:18.35805937+03:00 [INFO] read_socket;
2025-04-22T18:30:18.358638266+03:00 [INFO] read_socket;
2025-04-22T18:30:18.358658724+03:00 [INFO] Using Visual { id: 35, colormap: 0, depth: 32 }
2025-04-22T18:30:18.358686566+03:00 [INFO] Creating colormap 58720259
2025-04-22T18:30:18.358742901+03:00 [INFO] containing git repository: None
2025-04-22T18:30:18.358830313+03:00 [INFO] read_socket;
2025-04-22T18:30:18.359063828+03:00 [INFO] read_socket;
2025-04-22T18:30:18.359085788+03:00 [INFO] new;
2025-04-22T18:30:18.359187708+03:00 [INFO] read_socket;
2025-04-22T18:30:18.359270001+03:00 [INFO] read_socket;
2025-04-22T18:30:18.359343407+03:00 [INFO] read_socket;
2025-04-22T18:30:18.359558718+03:00 [INFO] read_socket;
2025-04-22T18:30:18.359596127+03:00 [INFO] keep_updated;
2025-04-22T18:30:18.359786291+03:00 [INFO] read_socket;
2025-04-22T18:30:18.359862623+03:00 [INFO] read_socket;
2025-04-22T18:30:18.359916603+03:00 [INFO] read_socket;
2025-04-22T18:30:18.365631068+03:00 [INFO] Using surface present mode FIFO_RELAXED
2025-04-22T18:30:18.407837861+03:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-04-22T18:30:18.407866725+03:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-04-22T18:30:18.407886251+03:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2025-04-22T18:30:18.422055004+03:00 [INFO] Refreshing at 6945 micros
2025-04-22T18:30:18.422123081+03:00 [INFO] x11: no compositor present, falling back to server-side window decorations
2025-04-22T18:30:18.423235319+03:00 [INFO] Using GPU: GpuSpecs { is_software_emulated: false, device_name: "NVIDIA GeForce RTX 4090", driver_name: "NVIDIA", driver_info: "570.86.10" }
2025-04-22T18:30:18.430091416+03:00 [INFO] XConnected server on 12582947, transport version: 0.0, TRANSPORT_MAX: 20
2025-04-22T18:30:18.430777823+03:00 [INFO] set status on client 0: Authenticating
2025-04-22T18:30:18.430929224+03:00 [INFO] Opening main db
2025-04-22T18:30:18.430952778+03:00 [INFO] perform;
2025-04-22T18:30:18.430978446+03:00 [INFO] authenticate;
2025-04-22T18:30:18.431015314+03:00 [INFO] read_command;
2025-04-22T18:30:18.431073292+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:18.431229983+03:00 [INFO] Opening main db
2025-04-22T18:30:18.431441877+03:00 [INFO] socket reader;
2025-04-22T18:30:18.43147004+03:00 [INFO] read_socket;
2025-04-22T18:30:18.431509413+03:00 [INFO] read_socket;
2025-04-22T18:30:18.43358402+03:00 [INFO] read_socket;
2025-04-22T18:30:18.434174718+03:00 [INFO] read_socket;
2025-04-22T18:30:18.434337922+03:00 [INFO] read_socket;
2025-04-22T18:30:18.434534617+03:00 [INFO] read_socket;
2025-04-22T18:30:18.434570304+03:00 [INFO] new;
2025-04-22T18:30:18.434676501+03:00 [INFO] read_socket;
2025-04-22T18:30:18.434732225+03:00 [INFO] read_socket;
2025-04-22T18:30:18.43478336+03:00 [INFO] read_socket;
2025-04-22T18:30:18.434964817+03:00 [INFO] read_socket;
2025-04-22T18:30:18.435000744+03:00 [INFO] keep_updated;
2025-04-22T18:30:18.435157806+03:00 [INFO] read_socket;
2025-04-22T18:30:18.435240179+03:00 [INFO] read_socket;
2025-04-22T18:30:18.435290894+03:00 [INFO] read_socket;
2025-04-22T18:30:18.450389886+03:00 [INFO] set status on client 196055: Connecting
2025-04-22T18:30:18.450469895+03:00 [INFO] open paths ["/home/pav/.config/zed/settings.json"]
2025-04-22T18:30:18.451416615+03:00 [INFO] Opening main db
2025-04-22T18:30:18.455314973+03:00 [INFO] containing git repository: None
2025-04-22T18:30:18.467075319+03:00 [INFO] attempting to start language server "json-language-server", path: "/home/pav/.config/zed/settings.json", id: 1
2025-04-22T18:30:18.467114782+03:00 [INFO] Initializing default prettier with plugins {}
2025-04-22T18:30:18.468656509+03:00 [INFO] Initializing default prettier with plugins {}
2025-04-22T18:30:18.469015857+03:00 [ERROR] failed to get git blame data: failed to find a git repository for buffer
2025-04-22T18:30:18.46965755+03:00 [INFO] Finish deserializing breakpoints & initializing breakpoint store
2025-04-22T18:30:18.479009157+03:00 [ERROR] failed to get git blame data: failed to find a git repository for buffer
2025-04-22T18:30:18.485164992+03:00 [INFO] Initializing default prettier with plugins {}
2025-04-22T18:30:18.512278604+03:00 [INFO] set environment variables from shell:/bin/bash, path:/home/pav/.opam/default/bin:/home/pav/.local/bin:/home/pav/yandex-cloud/bin:/home/pav/.opam/default/bin:/home/pav/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pav/go/bin:/home/pav/.cargo/bin:/home/pav/.cabal/bin:/home/pav/.ghcup/bin:/home/pav/go/bin:/home/pav/.cargo/bin
2025-04-22T18:30:18.591498203+03:00 [INFO] Initializing default prettier with plugins {}
2025-04-22T18:30:18.623276838+03:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2025-04-22T18:30:18.637943667+03:00 [INFO] using project environment variables shell launched in "/home/pav/.config/zed". PATH="/home/pav/.opam/default/bin:/home/pav/.local/bin:/home/pav/yandex-cloud/bin:/home/pav/.opam/default/bin:/home/pav/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pav/go/bin:/home/pav/.cargo/bin:/home/pav/.cabal/bin:/home/pav/.ghcup/bin:/home/pav/go/bin:/home/pav/.cargo/bin"
2025-04-22T18:30:18.638052359+03:00 [INFO] fetching latest version of language server "json-language-server"
2025-04-22T18:30:18.655315045+03:00 [INFO] using project environment variables shell launched in "/home/pav/temp/zedpanic". PATH="/home/pav/.opam/default/bin:/home/pav/.local/bin:/home/pav/yandex-cloud/bin:/home/pav/.opam/default/bin:/home/pav/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pav/go/bin:/home/pav/.cargo/bin:/home/pav/.cabal/bin:/home/pav/.ghcup/bin:/home/pav/go/bin:/home/pav/.cargo/bin"
2025-04-22T18:30:19.028297105+03:00 [INFO] Initializing default prettier with plugins {}
2025-04-22T18:30:19.197866058+03:00 [INFO] add_connection;
2025-04-22T18:30:19.2172729+03:00 [INFO] set status on client 196055: Connected { peer_id: PeerId { owner_id: 734, id: 1038247 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2025-04-22T18:30:19.400814742+03:00 [INFO] language server "json-language-server" is already installed
2025-04-22T18:30:19.400895382+03:00 [INFO] starting language server process. binary path: "/snap/bin/node", working directory: "/home/pav/.config/zed", args: ["/home/pav/.local/share/zed/languages/json-language-server/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server", "--stdio"]
2025-04-22T18:30:19.454181565+03:00 [INFO] perform;
2025-04-22T18:30:19.454225066+03:00 [INFO] authenticate;
2025-04-22T18:30:19.454256394+03:00 [INFO] perform;
2025-04-22T18:30:19.454284306+03:00 [INFO] authenticate;
2025-04-22T18:30:19.454307178+03:00 [INFO] perform;
2025-04-22T18:30:19.454333878+03:00 [INFO] authenticate;
2025-04-22T18:30:19.454354877+03:00 [INFO] perform;
2025-04-22T18:30:19.454390123+03:00 [INFO] authenticate;
2025-04-22T18:30:19.454417604+03:00 [INFO] perform;
2025-04-22T18:30:19.454450625+03:00 [INFO] authenticate;
2025-04-22T18:30:19.454483396+03:00 [INFO] perform;
2025-04-22T18:30:19.454521246+03:00 [INFO] authenticate;
2025-04-22T18:30:19.454554699+03:00 [INFO] read_command;
2025-04-22T18:30:19.45458797+03:00 [INFO] read_command;
2025-04-22T18:30:19.454613598+03:00 [INFO] read_command;
2025-04-22T18:30:19.454632533+03:00 [INFO] read_command;
2025-04-22T18:30:19.454658922+03:00 [INFO] read_command;
2025-04-22T18:30:19.454679801+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:19.454699488+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:19.454731467+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:19.454762064+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:19.454787622+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:19.454817567+03:00 [INFO] read_command;
2025-04-22T18:30:19.454864625+03:00 [INFO] send_secondary_commands;
2025-04-22T18:30:19.455114029+03:00 [INFO] socket reader;
2025-04-22T18:30:19.455132663+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455220647+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455240424+03:00 [INFO] socket reader;
2025-04-22T18:30:19.455260611+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455300125+03:00 [INFO] socket reader;
2025-04-22T18:30:19.455320132+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455344617+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455366698+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455415579+03:00 [INFO] socket reader;
2025-04-22T18:30:19.455433272+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455452127+03:00 [INFO] socket reader;
2025-04-22T18:30:19.455472425+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455495077+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455515455+03:00 [INFO] socket reader;
2025-04-22T18:30:19.455536574+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455558806+03:00 [INFO] read_socket;
2025-04-22T18:30:19.455578212+03:00 [INFO] read_socket;
2025-04-22T18:30:19.457139915+03:00 [INFO] read_socket;
2025-04-22T18:30:19.457554837+03:00 [INFO] Initializing default prettier with plugins {}
2025-04-22T18:30:19.458593187+03:00 [INFO] read_socket;
2025-04-22T18:30:19.460087796+03:00 [INFO] read_socket;
2025-04-22T18:30:19.461390348+03:00 [INFO] read_socket;
2025-04-22T18:30:19.462766517+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464174123+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464207395+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464249764+03:00 [INFO] read_socket;
2025-04-22T18:30:19.46427989+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464311008+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464337337+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464363526+03:00 [INFO] read_socket;
2025-04-22T18:30:19.46438756+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464409892+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464440328+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464464273+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464503296+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464546406+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464634109+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464656981+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464684913+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464845281+03:00 [INFO] read_socket;
2025-04-22T18:30:19.464953082+03:00 [INFO] read_socket;
2025-04-22T18:30:19.465065561+03:00 [INFO] read_socket;
2025-04-22T18:30:20.819504053+03:00 [ERROR] {
  "thread": "main",
  "payload": "begin <= end (10 <= 1) when slicing ` '{\"123\"}'\n`",
  "location_data": {
    "file": "/home/ubuntu/actions-runner/_work/zed/zed/crates/rope/src/chunk.rs",
    "line": 158
  },
  "backtrace": [
    "zed::reliability::init_panic_hook::{{closure}}::hba4876f7a43de794+141243845",
    "std::panicking::rust_panic_with_hook::h541791bcc774ef34+115626131",
    "std::panicking::begin_panic_handler::{{closure}}::h6479a2f0137c7d19+115625354",
    "std::sys::backtrace::__rust_end_short_backtrace::ha04e7c0fc61ded91+115618873",
    "rust_begin_unwind+115624493",
    "core::panicking::panic_fmt::h5764ee7030b7a73d+8655904",
    "core::str::slice_error_fail_rt::h95fb5a3efddd85ad+40357868",
    "core::str::slice_error_fail::hbde71130ea7318aa+8657274",
    "rope::Cursor::summary::hf60c270157d54d20+48470125",
    "editor::selections_collection::resolve_selections_display::he6ebf8f23d83e820+51321098",
    "editor::selections_collection::resolve_selections::h022c50a9fa424f02+51324369",
    "editor::selections_collection::SelectionsCollection::all::h0b7bb5f62b191ddc+51248872",
    "editor::Editor::selections_did_change::hb0dc42b0312fa416+51434157",
    "vim::normal::change::<impl vim::Vim>::change_object::h05457e50c673136d+127153736",
    "vim::object::<impl vim::Vim>::object::hf2635a15c85d7853+127387006",
    "editor::Editor::register_action::{{closure}}::{{closure}}::h1672d8eb8f478874+126442042",
    "gpui::window::Window::dispatch_action_on_node::h368b2c26dccb3c81+68529231",
    "gpui::window::Window::dispatch_event::he7e723fcfe8dfd93+68604093",
    "gpui::window::Window::new::{{closure}}::h73bfdc5e77f53db9+68461949",
    "gpui::platform::linux::x11::window::X11WindowStatePtr::handle_input::h3518550db0c4249b+68194518",
    "gpui::platform::linux::x11::client::X11Client::handle_event::h12f8f42368c720a1+68130693",
    "gpui::platform::linux::x11::client::X11Client::process_x11_events::h62d886ef42a61e87+68107612",
    "<core::cell::RefCell<calloop::sources::DispatcherInner<S,F>> as calloop::sources::EventDispatcher<Data>>::process_events::h779ef8ed599a96a4+64763595",
    "<gpui::platform::linux::x11::client::X11Client as gpui::platform::linux::platform::LinuxClient>::run::h471a6e027a358cc9+68171143",
    "gpui::platform::linux::platform::<impl gpui::platform::Platform for P>::run::h07c2fde61651396f+67770173",
    "gpui::app::Application::run::hf58d124b652f6fbc+139277129",
    "zed::main::h5b80b3ef210555eb+141272007",
    "std::sys::backtrace::__rust_begin_short_backtrace::hd07c512352b21c1e+141139299",
    "std::rt::lang_start::{{closure}}::hdc43a22f0193ee17+142531529",
    "std::rt::lang_start_internal::h15895544e2012228+115549553",
    "main+141290220",
    "__libc_start_call_main+16339053179338",
    "__libc_start_main_impl+16339053179531",
    "_start+10930590"
  ],
  "app_version": "0.182.11",
  "app_commit_sha": "67ca67273f4066b703e3bed9d0a1a08b13010c43",
  "release_channel": "stable",
  "target": "x86_64-unknown-linux-gnu",
  "os_name": "Linux X11",
  "os_version": "ubuntu 24.04",
  "architecture": "x86_64",
  "panicked_on": 1745335820819,
  "system_id": "39fe4d6c-0827-4706-b8b3-3318e83b0705",
  "installation_id": "ca5e4912-c070-45a9-9419-90ca8235fe76",
  "session_id": "57a680fe-29e2-473b-a60a-22e9e895b85f"
}
```

</details>

--- a/crates/languages/src/bash/brackets.scm
+++ b/crates/languages/src/bash/brackets.scm
@@ -3,4 +3,3 @@
 ("{" @open "}" @close)
 ("\"" @open "\"" @close)
 ("`" @open "`" @close)
-((raw_string) @open @close)

